base/broadcast.jl

```diff
@@ -279,8 +305,16 @@ ziptype(A) = typestuple(A)
 ziptype(A, B) = (Base.@_pure_meta; Iterators.Zip2{typestuple(A), typestuple(B)})
 @inline ziptype(A, B, C, D...) = Iterators.Zip{typestuple(A), ziptype(B, C, D...)}
 
-_broadcast_type(f, T::Type, As...) = Base._return_type(f, typestuple(T, As...))
-_broadcast_type(f, A, Bs...) = Base._default_eltype(Base.Generator{ziptype(A, Bs...), ftype(f, A, Bs...)})
+scalarify_nullable(x) = x
+scalarify_nullable(x::Nullable) = (x,)
```

Comment:
Instead all of this you could only define `typestuple(a::Nullable) = (Base.@_pure_meta; Tuple{typeof(a)})` above whitout modifying the `btroadcast_type` methods.